src/test/kotlin/com/mparticle/kits/AppboyKitTest.kt

```diff
@@ -446,7 +449,288 @@ class AppboyKitTests {
+        if (productArray is Array<*>) {
+            Assert.assertEquals(1, productArray.size.toLong())
+            val productBrazeProperties = productArray[0]
+            if (productBrazeProperties is BrazeProperties) {
+                val productProperties = productBrazeProperties.properties
+                Assert.assertEquals(productProperties.remove(CommerceEventUtils.Constants.ATT_PRODUCT_TOTAL_AMOUNT), 22.5)
+                Assert.assertEquals(productProperties.remove(CommerceEventUtils.Constants.ATT_PRODUCT_PRICE), 4.5)
+                Assert.assertEquals(productProperties.remove(CommerceEventUtils.Constants.ATT_PRODUCT_QUANTITY), 5.0)
+                Assert.assertEquals(productProperties.remove(CommerceEventUtils.Constants.ATT_PRODUCT_ID), "sku1")
+                Assert.assertEquals(productProperties.remove(CommerceEventUtils.Constants.ATT_PRODUCT_NAME), "product name")
+                Assert.assertEquals(emptyAttributes, productProperties)
+            }
+        }
+        Assert.assertEquals(properties.remove(CommerceEventUtils.Constants.ATT_SHIPPING), 12.0)
+        Assert.assertEquals(properties.remove(CommerceEventUtils.Constants.ATT_TAX), 100.0)
+        Assert.assertEquals(
+            properties.remove(CommerceEventUtils.Constants.ATT_PRODUCT_COUPON_CODE),
+            "coupon code"
+        )
+        Assert.assertEquals(
+            properties.remove(CommerceEventUtils.Constants.ATT_AFFILIATION),
+            "the affiliation"
+        )
+        Assert.assertEquals(
+            properties.remove(CommerceEventUtils.Constants.ATT_ACTION_PRODUCT_LIST_SOURCE),
+            "the source"
+        )
+        Assert.assertEquals(properties.remove(CommerceEventUtils.Constants.ATT_TOTAL), 99.0)
+        Assert.assertEquals(
+            properties.remove(CommerceEventUtils.Constants.ATT_ACTION_PRODUCT_ACTION_LIST),
+            "product list name"
+        )
+        Assert.assertEquals(
+            properties.remove(CommerceEventUtils.Constants.ATT_TRANSACTION_ID),
+            "the id"
+        )
+        Assert.assertEquals(properties.remove("key1"), "value1")
+        Assert.assertEquals(properties.remove("key #2"), "value #3")
+        Assert.assertEquals(emptyAttributes, properties)
+    }
+
+    @Test
+    fun testPromotion() {
+        val kit = MockAppboyKit()
+        kit.configuration = MockKitConfiguration()
+        val customAttributes = HashMap<String, String>()
+        customAttributes["key1"] = "value1"
+        customAttributes["key #2"] = "value #3"
+        val promotion = Promotion().apply {
+            id = "my_promo_1"
+            creative = "sale_banner_1"
+            name = "App-wide 50% off sale"
+            position ="dashboard_bottom"
+        }
+        val commerceEvent = CommerceEvent.Builder(Promotion.VIEW, promotion)
+            .customAttributes(customAttributes)
+            .build()
+        val eventList = CommerceEventUtils.expand(commerceEvent)
```

Comment:
Should you be expanding this here?  I thought that logEvent will do the expansion.  https://github.com/mparticle-integrations/mparticle-android-integration-appboy/blob/e914f26543de2bc612fb8f3206cf9b372e4db590/src/main/kotlin/com/mparticle/kits/AppboyKit.kt#L226